.buildkite/release-pipelines/publish-release.yml

```diff
@@ -1,21 +1,26 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
+---
+
+agents:
+  queue: "mac"
+
+env:
+  IMAGE_ID: $IMAGE_ID
+
 steps:
   - label: "Publish Release"
-    plugins:
-      - $CI_TOOLKIT
+    plugins: [$CI_TOOLKIT]
     command: |
-      echo '--- :robot_face: Use bot for git operations'
-      source use-bot-for-git
+      .buildkite/commands/configure-environment.sh
 
       echo '--- :git: Checkout Release Branch'
       .buildkite/commands/checkout-release-branch.sh
 
-      echo '--- :ruby: Setup Ruby Tools'
-      install_gems
+      echo '--- :closed_lock_with_key: Access Secrets'
+      bundle exec fastlane run configure_apply
 
       echo '--- :package: Publish Release'
       bundle exec fastlane publish_release skip_confirm:true
-    agents:
-      queue: "tumblr-metal"
```

Comment:
Why the change from `tumblr-metal` to the `mac` agents?

Reply:
The point of using `tumblr-metal` for release tasks is to allow for them to run on a trusted agent that has the proper credentials (via using `use-bot-for-git`).
Unfortunately, the solution doesn't work for some macOS release tasks that need macOS tooling (e.g. to generate strings).

On this specific case, it actually wouldn't make a difference because the lane `publish_release` doesn't run anything that requires a Mac, so I guess it would just work on `tumblr-metal` (but without using `use-bot-for-git`, as it only supports a pre-defined set of repos -- see paaHJt-6EP-p2).

To make it consistent with the other release pipelines I decided to let it run on the same `mac` queue for now, until we build a Mac solution for trusted agents and all release tasks can run without needing a repo deploy key.

Reply:
@iangmaia I'm not sure I follow.

As long as the `wpmobilebot` (whose SSH key is already installed in the trusted tumblr-metal agent) also has access to the `woocommerce-ios` repo (just like it already has access to `woocommerce-android`, we should be able to use `tumblr-metal` in this pipeline and call `use-bot-for-git wpmobilebot` to tell it to use that bot's key explicitly and this should work, no?

Sure, `woocommerce-ios` is [not one of the repos for which our `use-bot-for-git` helper would auto-detect the name of the bot user and SSH key to use](https://github.com/Automattic/buildkite-ci/blob/trunk/src/agents/tumblr-metal/resources/use-bot-for-git.sh#L40), but the whole reason why [we added the possibility to pass an extra parameter to this script](https://github.com/Automattic/buildkite-ci/blob/trunk/src/agents/tumblr-metal/resources/use-bot-for-git.sh#L35-L37) was to be able to support other repos that were not yet in the auto-detect list… without having to redeploy the `tumblr-metal` agent every time we wanted to use it for a new repo, right?

Reply:
Had forgotten about the `use-bot-for-git` parameter option, so yes it would work as it doesn't have anything macOS specific as I mentioned.
At the same time, my main goal with the change was to keep it consistent with all the other release pipelines -- even when we eventually have a trusted agent solution for iOS, it would always be puzzling why this single one is running on a different agent.

Reply:
I mean I get it, but running steps that require git push access on `tumblr-metal` even on a iOS/Mac pipeline is already what we do on other repos (recent case in point: https://github.com/Automattic/simplenote-ios/pull/1650), so… 🤷